packages/framework/src/Model/Mail/Mailer.php

```diff
@@ -5,20 +5,54 @@
 use Swift_Attachment;
 use Swift_Mailer;
 use Swift_Message;
+use Symfony\Component\HttpKernel\KernelInterface;
 
 class Mailer
 {
+    /**
+     * @var \Symfony\Component\HttpKernel\KernelInterface
+     */
+    protected $kernel;
+
     /**
      * @var \Swift_Mailer
      */
     protected $swiftMailer;
 
     /**
      * @param \Swift_Mailer $swiftMailer
+     * @param \Symfony\Component\HttpKernel\KernelInterface $kernel
      */
-    public function __construct(Swift_Mailer $swiftMailer)
+    public function __construct(Swift_Mailer $swiftMailer, KernelInterface $kernel)
     {
         $this->swiftMailer = $swiftMailer;
+        $this->kernel = $kernel;
+    }
+
+    public function flushSpoolQueue()
+    {
+        $container = $this->kernel->getContainer();
+        if (!$container->has('mailer')) {
+            return;
+        }
+        $mailers = array_keys($container->getParameter('swiftmailer.mailers'));
+        foreach ($mailers as $name) {
+            if (method_exists($container, 'initialized') ? $container->initialized(sprintf('swiftmailer.mailer.%s', $name)) : true) {
```

Comment:
I think this could be more understandable if it was in its own protected function/functions (maybe with the second `if` as well)

Reply:
hold-on EmailSenderListener.php::onTerminate